CAN_Decoder_SourceCode_TestRel_21Dec21.py

```diff
@@ -0,0 +1,119 @@
+import tkinter as tk
+from tkinter import *
+from tkinter import filedialog
+from datetime import datetime
+import re
+
+
+def CAN_Decoder():
+    dataFile = filedialog.askopenfilename()
+    inputFrameData = []
+    
+    # Extract frame from text file using regex approach
+    typicalFrameForm = r"can(\d)\s+([(\w)]{1,3})\s+\[(\d+)\]\s+([\w]{2}(?:\s[\w]{2})*)"
+    frameSize = 0
+    for dataFrame in re.findall(typicalFrameForm, open(dataFile).read()):
+        diagnosticID = int(dataFrame[1], 16)
+        data = list(map(lambda x: int(x, 16), dataFrame[3].split(" ")))
+        frameType = data[0]
+        if frameType <= 6:
```

Comment:
This is right for this example, but for can messages larger than 8 bytes, this will not work.
The check should be `data[0] // 16 == 0`